### DISCUSSION TITLE
How to Use it with FastAPI Server

### DISCUSSION THREAD

**Discussion by Gmonga08** [reflex-dev/reflex#488] [Q&A]
Hello Everyone, 
Can Anyone Tell me How to use Pynecone with My Server That uses FastAPI? 
I already Developed Server Using Fast API, Now i want to create a frontend in either way :-
Non Web Based - using pyside6 
Or Web Based - using pynecone ( i dont know how to code in Javascript ).

**Comment by FHU-yezi** [comment]
Now you are able to add custom API routes to your Pynecone App, but I'm not sure if you can got the low-level FastAPI App object and combine it with your exist FastAPI App.

Use a reverse proxy should work.

**Reply by Lendemor** [reply]
> Now you are able to add custom API routes to your Pynecone App, but I'm not sure if you can got the low-level FastAPI App object and combine it with your exist FastAPI App.
> 
> Use a reverse proxy should work.

It's not something I've tested so not guaranteed, but I think it should be possible to import an existing FastAPI app and mount it as a sub-app on the one used by Pynecone (`app.api`).

If not that, running them as separate instance and sending API query from Pynecone to their own backend is possible. (with reverse proxy if they are on different machines).
